src/Pinch/Generate.hs

```diff
@@ -159,10 +159,43 @@ gInclude dir i = do
 
 gDefinition :: Definition SourcePos -> GenerateM ([H.Decl], [H.Decl], [H.Decl])
 gDefinition def = case def of
-  ConstDefinition _ -> pure ([], [], [])
+  ConstDefinition c -> (\x -> (x, [], [])) <$> gConst c
   TypeDefinition ty -> (\x -> (x, [], [])) <$> gType ty
   ServiceDefinition s -> gService s
 
+gConst :: A.Const SourcePos -> GenerateM [H.Decl]
+gConst const = do
+  tyRef <- gTypeReference (constValueType const)
+  value <- gConstValue (constValue const)
+  pure
+    [ H.TypeSigDecl name tyRef
+    , H.FunBind [H.Match name [] value]
+    ]
+  where
+    name = decapitalize (constName const)
+
+gConstValue :: A.ConstValue SourcePos -> GenerateM H.Exp
+gConstValue val = case val of
+  ConstInt n _ -> pure (H.ELit (H.LInt n))
+  ConstFloat n _ -> pure (H.ELit (H.LFloat n))
+  ConstLiteral s _ -> pure (H.ELit (H.LString s))
+  ConstIdentifier id _
+    | xs @(_:_:_) <- T.splitOn "." id -> do
+      map <- asks cModuleMap
+      case Map.lookup (mconcat $ init xs) map of
+        Nothing ->
+          -- TODO this should probably be an error
+          pure (H.ELit (H.LRef Nothing (decapitalize id)))
+        Just mod ->
+          pure $ H.ELit (H.LRef (Just mod) (decapitalize (last xs)))
+    | otherwise -> pure $ H.ELit (H.LRef Nothing (decapitalize id))
```

Comment:
On second thoughts, I think we can actually use H.EVar here and avoid the LRef constructor entirely. I think that would be a bit nicer as Haskell does not really have "references" as literals. I think then we are good to go.